src/common/translations/blank_before_unit.ts

```diff
@@ -3,7 +3,7 @@ import { blankBeforePercent } from "./blank_before_percent";
 
 export const blankBeforeUnit = (
   unit: string,
-  localeOptions?: FrontendLocaleData
+  localeOptions: FrontendLocaleData | undefined
```

Comment:
This is the same thing.  Why do we need to handle `undefined`?

Reply:
Many of the classes that call this have optionally defined locale or hass.

I might be misunderstanding but I thought this was slightly better, as it requires the locale option to be passed, even if it undefined. So I believe it may be an improvement?

Reply:
Oh right... `hass` is non-null asserted in like ~10% of components - yet another inconsistency to fix. 🤷🏻‍♂️ 

And these functions should be removed anyway as we should be using `Intl.NumberFormat` for this purpose.